plugins/vault-backend/src/service/VaultBuilder.ts

```diff
@@ -101,25 +104,32 @@ export class VaultBuilder {
   }
 
   /**
-   * Enables the token renewal for Vault.
+   * Enables the token renewal for Vault. The schedule if configured in the app-config.yaml file.
+   * If not set, the renewal is executed hourly
    *
-   * @param schedule - The TaskRunner used to schedule the renewal, if not set, renewing hourly
    * @returns
    */
-  public async enableTokenRenew(schedule?: TaskRunner) {
-    const taskRunner = schedule
-      ? schedule
-      : this.env.scheduler.createScheduledTaskRunner({
+  public async enableTokenRenew() {
```

Comment:
If feel we could keep the existing schedule option intact here tbh to avoid the breaking change, doesn't hurt much and it'll eventually be removed as we remove support for the old system from this plugin. So the logic would be that it uses the passed in schedule if present, then checks config, and then falls back to the default.